virtualmachine/gcp/vm.go

```diff
@@ -56,17 +65,127 @@ type VM struct {
 
 	AccountFile  string
 	account      accountFile
-	SSHCreds     ssh.Credentials // privateKey is required for GCE
+	SSHCreds     ssh.Credentials
 	SSHPublicKey string
+
+	Firewall  string // required when modifying firewall rules
```

Comment:
is it the rule-name?

Reply:
This naming in ambiguous in GCP. On console, it is called as Firewall Rule, but in other documentation and all the APIs, it is called [Firewall](https://cloud.google.com/compute/docs/reference/latest/firewalls#resource).